crates/ty_python_semantic/src/types/class.rs

```diff
@@ -3624,13 +3656,12 @@ impl KnownClass {
         }
     }
 
-    /// Return `true` if this class is a [`SolidBase`]
-    const fn is_solid_base(self) -> bool {
+    /// Return `true` if this class is a [`DisjointBase`]
+    const fn is_disjoint_base(self) -> bool {
```

Comment:
In principle we could eliminate this method and always look at the typeshed fallback class (and its decorator), right? This is just a small performance optimization (maybe) at the cost of preserving some special-casing that we technically shouldn't need anymore?